Add TS decorator example

PeterStaev opened the pull request:
Relates to https://github.com/NativeScript/ios-runtime/pull/753

ivanbuhov commented:
@PeterStaev Since we still support subclassing without decorators lets not remove it entirely from the doc. I suggest leaving the TS example as is and adding another section next or under the "TypeScript support" section documenting the decorators approach.

PeterStaev commented:
@ivanbuhov totally agree! My mistake that I replaced existing examples. And moreover users might be using a pre-3.0 version where there are no decorators. Adjusted now.
